cmd/snapd-apparmor/main_test.go

```diff
@@ -151,6 +159,12 @@ func (s *mainSuite) TestIsContainer(c *C) {
 	c.Check(snapd_apparmor.IsContainer(), Equals, false)
 	c.Assert(detectCmd.Calls(), DeepEquals, [][]string{
 		{"systemd-detect-virt", "--quiet", "--container"}})
+
+	// Test WSL2 with custom kernel
```

Comment:
Maybe I miss context here but what is the relation between "systemd-detect-virt" failing with "failed" and that WSL2 uses a custom kernel :) ? Maybe the comment could explain this a little bit more (e.g. `// Test WSL2 with custom kernel where systemd-detect-virt will return a non-zero exit code as it does not consider it a container`) (assuming this is what is happening here?) and (nitpick^2) maybe instead of "failed" the mock systemd-detect-virlt could print something like "none" (assuming this is what the real systemd-detect-virt would return on a WSL2+custom-kernel)?

Reply:
You're right, `echo none && exit 1` is the correct behaviour. Fixed.